lib/rspec/matchers/built_in/change.rb

```diff
@@ -87,8 +87,8 @@ def initialize(receiver=nil, message=nil, &block)
         end
 
         def raise_block_syntax_error
-          raise SyntaxError, "The block passed to the `change` matcher must " \
-            "use `{ ... }` instead of do/end"
+          raise SyntaxError, "Block passed to the wrong method. Perhaps you " \
```

Comment:
If someone does this: `expect{ subject }.to(change{}){ some_query.count }.by(1)` they can expect a weird message because they're doing something completely unsupported, the goal is to provide a clearer message to the majority of use cases, which assumes one block attempting to be passed to the `change` matcher.